packages_rs/nextclade/src/tree/tree_builder.rs

```diff
@@ -106,23 +107,26 @@ pub fn finetune_nearest_node(
   graph: &AuspiceGraph,
   nearest_node_key: GraphNodeKey,
   seq_private_mutations: &BranchMutations,
+  params: &TreeBuilderParams,
 ) -> Result<(GraphNodeKey, BranchMutations), Report> {
+  let masked_ranges = graph.data.meta.placement_mask_ranges();
+
   let mut best_node = graph.get_node(nearest_node_key)?;
   let mut private_mutations = seq_private_mutations.clone();
 
   loop {
     // Check how many mutations are shared with the branch leading to the current_best_node or any of its children
-    let (candidate_node, candidate_split, n_shared_muts) = find_shared_muts(graph, best_node, &private_mutations)
-      .wrap_err_with(|| {
+    let (candidate_node, candidate_split, shared_muts_score) =
+      find_shared_muts(graph, best_node, &private_mutations, masked_ranges, params).wrap_err_with(|| {
         format!(
           "When calculating shared mutations against the current best node '{}'",
           best_node.payload().name
         )
       })?;
 
     // Check if the new candidate node is better than the current best
-    let n_left_muts = count_nuc_muts(&candidate_split.left.nuc_muts);
-    match find_better_node_maybe(graph, best_node, candidate_node, n_shared_muts, n_left_muts) {
+    let left_muts_score = score_nuc_muts(&candidate_split.left.nuc_muts, masked_ranges, params);
+    match find_better_node_maybe(graph, best_node, candidate_node, shared_muts_score, left_muts_score) {
```

Comment:
I use the new score for both "shared" and "left" muts, even though "left" is currenly only used to compare to 0 (i.e. it is equivalent to `.is_empty()`)